hazelcast/src/main/java/com/hazelcast/ringbuffer/impl/RingbufferService.java

```diff
@@ -303,4 +328,130 @@ private void clearRingbuffersHavingLesserBackupCountThan(int partitionId, int th
     public boolean isKnownServiceNamespace(ServiceNamespace namespace) {
         return namespace instanceof ObjectNamespace;
     }
+
+    @Override
+    public Runnable prepareMergeRunnable() {
+        Map<Integer, Map<ObjectNamespace, RingbufferContainer>> partitionContainerMap = createHashMap(containers.size());
+        IPartitionService partitionService = nodeEngine.getPartitionService();
+        Address thisAddress = nodeEngine.getClusterService().getThisAddress();
+
+        for (Entry<Integer, Map<ObjectNamespace, RingbufferContainer>> entry : containers.entrySet()) {
```

Comment:
At this point I really feel like I should have created a separate class for the partition map. That way this per-partition logic could be moved to that class as well. Maybe we can refactor this later.